### PR TITLE
Fix: Close leak.

### PR DESCRIPTION
**Pull request by kimkulling**
- closes https://github.com/assimp/assimp/issues/3416

### CODE REVIEW

File: include/assimp/MemoryIOWrapper.h [hunk 1]
```diff
@@ -162,7 +157,11 @@ class MemoryIOSystem : public IOSystem {
     }
 
     /** Destructor. */
-    ~MemoryIOSystem() = default;
+    ~MemoryIOSystem() {
+        for (auto &it : created_streams) {
+            MemoryIOSystem::Close(it);
```
Comment (turol) [inline include/assimp/MemoryIOWrapper.h:162]: `MemoryIOSystem::Close` modifies `created_streams` while the destructor is iterating over it.

**Comment by turol** [comment]
Try something like this:
```
while (!created_streams.empty()) {
  MemoryIOSystem::Close(created_streams.front());
}
```
Not very nice but probably the best we can do since the destructor has to remove itself from the map

**Comment by kimkulling** [comment]
It seems that there is a mismatch in the way other importers aree closing the files. I will check it this weekend.

**Comment by kimkulling** [comment]
Seems not to fix the issue.

**Pull request closed without merging**